test/hotspot/jtreg/compiler/lib/template_framework/TemplateFrame.java

```diff
@@ -27,38 +27,70 @@
 import java.util.Map;
 
 /**
- * The {@link TemplateFrame} is the frame for a {@link Template}, i.e. the corresponding
- * {@link TemplateToken}. It ensures that each template use has its own unique {@link #id}
- * used to deconflict names using {@link Template#$}. It also has a set of hashtag
+ * The {@link TemplateFrame} is the frame for a {@link Template} and its inner
+ * {@link Template#scope}s. It ensures that each {@link Template} use has its own unique
+ * {@link #id} used to deconflict names using {@link Template#$}. It also has a set of hashtag
  * replacements, which combine the key-value pairs from the template argument and the
- * {@link Template#let} definitions. The {@link #parent} relationship provides a trace
- * for the use chain of templates. The {@link #fuel} is reduced over this chain, to give
- * a heuristic on how much time is spent on the code from the template corresponding to
- * the frame, and to give a termination criterion to avoid nesting templates too deeply.
+ * {@link Template#let} definitions. Inner scopes of a {@link Template} have access to
```

Comment:
`Inner scopes of a {@link Template} [...]`

Maybe add something here that we only mean scopes that are not template themselves.

Reply:
I don't really know how to say it better than I am already. Do you have a suggestion?